spec/features/users/my_spec.rb

```diff
@@ -124,115 +124,303 @@ def expect_changed!
+        it 'shows notice about disabled token' do
+          visit my_access_token_path
 
-          within(:xpath, "//tr[contains(.,'#{expected_content}')]") do
-            expect(page).to have_content expected_content
-            expect(page).to have_content I18n.l(another_ical_token_for_query.created_at, format: :time).to_s
+          within '#icalendar-token-section' do
+            expect(page).to have_content('iCalendar subscriptions are not enabled by the administrator.')
           end
         end
+      end
 
-        Timecop.travel(2.minutes.from_now) do
-          ical_token_for_another_query = create(:ical_token, user:, query: another_query, name: "Some Token Name")
+      context 'when iCalendar access is enable via global settings', with_settings: { ical_enabled: true } do
+        context 'when no iCalendar token exists' do
+          it 'shows notice about how to use iCalendar tokens' do
+            visit my_access_token_path
 
-          visit my_access_token_path
+            within '#icalendar-token-section' do
+              expect(page).to have_content('To add an iCalendar token') # ...
+            end
+          end
+        end
+
+        context 'when multiple iCalendar tokens exist' do
+          let!(:project) { create(:project) }
+          let!(:query) { create(:query, project:) }
+          let!(:another_query) { create(:query, project:) }
+          let!(:ical_token_for_query) { create(:ical_token, user:, query:, name: "First Token Name") }
+          let!(:ical_token_for_another_query) { create(:ical_token, user:, query: another_query, name: "Second Token Name") }
+          let!(:second_ical_token_for_query) { create(:ical_token, user:, query:, name: "Third Token Name") }
+
+          it 'shows iCalendar tokens with their calender and project info' do
+            visit my_access_token_path
+
+            expect(page).not_to have_content('To add an iCalendar token') # ...
+
+            within '#icalendar-token-section' do
+              [
+                ical_token_for_query,
+                ical_token_for_another_query,
+                second_ical_token_for_query
+              ].each do |ical_token|
+                token_name = ical_token.ical_token_query_assignment.name
+                query = ical_token.ical_token_query_assignment.query
+                row_xpath = "//tr[td[1][contains(text(), '#{token_name}')]]"
+
+                expect(page).to have_xpath("#{row_xpath}/td[1]", text: token_name)
+                expect(page).to have_xpath("#{row_xpath}/td[2]/a", text: query.name)
+                expect(page).to have_xpath("#{row_xpath}/td[3]", text: query.project.name)
+              end
+            end
+          end
+
+          it 'single iCalendar tokens can be deleted' do
+            visit my_access_token_path
+            token_name = ical_token_for_query.ical_token_query_assignment.name
+            row_xpath = "//tr[td[1][contains(text(), '#{token_name}')]]"
```

Comment:
To avoid complicated statement like this, we introduced data attributes to be able to more easily target a DOM node in the specs: `data-qa-selector`